File: backend/gatorsched_api/schemas/manager/shifts/edit_shift.py
from datetime import date

from pydantic import BaseModel

from gatorsched_api.schemas.manager.shifts.shared_types import EditedShift, TimeRange


class EditShiftRequest(BaseModel):
    date: date
    startTime: TimeRange
    endTime: TimeRange
    staffingRequirement: int
    roleId: str


# class EditedShift(BaseModel):
#     id: str
#     date: date
#     dayKey: str
#     shortDayLabel: str
#     longDayLabel: str
#     startTime: TimeRange
#     endTime: TimeRange
#     staffingRequirement: int
#     roleId: str
#     roleName: str
#     roleColor: str
#     fromTime: str
#     toTime: str


class EditShiftResponse(BaseModel):
    shift: EditedShift

File: frontend/src/services/shifts/editShift.ts
import { API_BASE } from '../apiConfig';
import { EditShiftRequest, EditedShift } from './types';

// export type EditShiftRequest = {
//   date: string;
//   startTime: TimeRange;
//   endTime: TimeRange;
//   staffingRequirement: number;
//   roleId: string;
// };

// export type EditedShift = {
//   id: string;
//   date: string;
//   dayKey: string;
//   shortDayLabel: string;
//   longDayLabel: string;
//   startTime: TimeRange;
//   endTime: TimeRange;
//   staffingRequirement: number;
//   roleId: string;
//   roleName: string;
//   roleColor: string;
//   fromTime: string;
//   toTime: string;
// };

export type EditShiftResponse = {
  shift: EditedShift;
};

export async function editShift(
  payload: EditShiftRequest,
  shiftId: string,
): Promise<EditShiftResponse> {
  const res = await fetch(`${API_BASE}/manager/shifts/${shiftId}`, {
    method: 'PUT',
    headers: {
      'Content-Type': 'application/json',
    },
    body: JSON.stringify(payload),
  });

  if (!res.ok) {
    throw new Error('Failed to edit Shift');
  }

  return res.json();
}
